---
permalink: /
title: "5+ yrs. professional"
subtitle: Matt Ralston - results-driven Research Associate
author_profile: true
hero_image: "mattralston.png"
redirect_from: 
  - /about/
  - /about.html
---

 Western/Northern blot, RT-PCR, GC-MS, HPLC, ligation, restriction enzymes, ELISA, IHC/IF, FACS-Aria, Illumina library preparation. You name it!

## Bioinformatics specialist

Hey there! I'm Matt, and I'm thrilled you've found your way to my corner of the internet. As a passionate bench biologist and bioinformatics programmer, I've created this space to share my work, ideas, and journey with you.
Feel free to explore using the Sitemap below to find exactly what you're looking for. And if you're an industry talent specialist or technical manager, I'd love for you to get to know me better â€“ head over to the 'About Me' section for a deeper dive into my background and expertise.


> "The scientific method is like a bicycle: you can go somewhere on it if you want to, but it's not really the sort of thing you can just play around with." - Terry Pratchett

## Sitemap

* - About
* [About Me - Matt Ralston : statement](https://matthewralston.github.io)
* - Subject areas
* [Biological Sciences](/biosciences)
* [Cheminformatics](/cheminformatics)
* [Research](/research)
* - Software projects - [Portfolio](/portfolio)
* [Software experience](/software)
* [Bioinformatics, Data Science, ML, AI, LLM](/data_science)
* - Blog - [Blog](/blog)
* - Social Media
* [LinkedIn](https://linkedin.com/in/MatthewRalston)
* [GitHub](https://github.com/MatthewRalston)
* - Resume / Curriculum Vitae (C.V.)
* [Resume/C.V.](/cv)
* - Personal Statement
* [Personal statement](/personal_statement)
* - Publications
* [Publications](https://scholar.google.com/citations?hl=en&pli=1&user=86Lxg-4AAAAJ)


## About Me

> Target audience: scientists, engineers, managers, and academic partners


Seeking a dynamic addition to your team? I bring 5 years of industry biopharmaceutical R&D team experience coupled with a decade of experience in wet-bench biology, Linux, and data analysis with Python/Perl/R. Let's elevate your projects together!

<!-- Are you looking for a new hire or academic collaborator with 5+ years of industry experience, 10+ years of bioinformatics, Linux, and Python/Perl/R coding expertise? -->

> Do you value culture, communication, productivity, and passion such that it guides your decision in hiring new talent?

> Do you require a professional adept at communicating in-betweena and bridging diverse disciplines? 

<!-- Do you need someone with the communication style needed for a melting-pot of scientists, engineers, and managerial professionals to be heard, understood, and valued? -->




 I work in groups of engineers, analysts, and wet-bench laboratory professionals, team leads, associate directors, and alike. I listen to as many voices as I can, and I share my experiences with others; I am detail oriented and enjoy the problem solving process, and I can write about it in a variety of ways throughout our journey. 
 
 

<!-- I am an *expert* at modeling, data-processing, and visualization techniques such as *R*, *Cython*/*Python*, *Javascript*, *Linux*/*bash*, *Docker*, *AWS*/cloud and more. It's more than just websites, APIs, numbers and figures here; my strong fundamentals and data-driven design strategy have already been used to affect diverse goals in industry teams: hypothesis generation, key-questions, application or integration needs, and more, such that results are easily reproducibile, documented, and clearly written. -->

So I do the bench science thing too! I'm effecient and effective and timely in the lab. Asepsis leads to less errors, and as a bench molecular biologist I'm familiar with the autoclave and where the imaging equipment is. I can perform single-gene targetted expression studies (sometimes referred to as qPCR or rt-PCR on mRNA molecules. I also work with Illumina lab reagent kit technologies). I do multiplex. I have some conceptual familiarity with a lot of different methods from chemical analytical, bioassay, immunoassay, fluorimetry, MS, GC-MS, LC-MS, ELISA, and an intimate understanding of cell culture and laminar flow hoods. My last acura was an Illumina but my last report was a Northern.

  * Regarding research - what you can expect.

I work independently with consistent email and Slack communications, and project updates and logging via Version Control and `git`. My project management software is often plain text, also git-based, metrics, org-files for Pandoc compatibility, second-brain, markdown where possible, license documents, source files and configurations, Dockerfiles where possible, and I write LaTeX primarily in `.Rmd`.


  * Culture:
  
  With some recommendations from industry managers, I've delivered value on projects within integrated teams of laboratory scientists, programmers, engineers, and managers on projects in wet-bench biology, bioinformatic and cheminformatic research areas. I take statistics, data, and modeling efforts seriously and I encourage you to check out the [software](/software), [research](/research), [biosciences](/biosciences), [LinkedIn](/linkedin), [testimonials](#testimonials) section and perhaps my modest GitHub [portfolio](/portfolio). I hope that you'll see I have some familiarity with publishing, only one first authorship formally. One open source project concerning sequence alignment conceptural understanding, k-mer and minimizer expertise, Illumina data profiling, and not caring what else needs work on besides my algorithm engineering training, and my R&D process.

  
  * Communication:

  With proficiencies in technical writing, science, engineering, and stats/probability/mathematics, I've delivered value and knowledge to others without losing touch. Your high-value team demands written style and know-how in publications, meetings, and documentation. Clear and compelling written style is essential to today's business successes, tomorrow's publications, and our quarterly progress. Check out the [publications](/publications) or [testimonials](#testimonials) sections for more reasons why I'm the better fit for your business needs.

  * Productivity:
  
  I use Rmarkdown and Python for most of my data science stack. I use Python SQLAlchemy, Flask, and Rust ORMs and RDBMS technologies to create normalized data structures during ETL. ETL is mostly Python, numpy, CUDA, and/or `rust` where I get the time to develop robustly and comfortably at my knowledge of static type function signatures and handling types robustly with GPT4.0-o1-mini in `zed` or Claude Sonnet 3.5+ mobile to assist with code when needed. I use minimal software stacks and don't do much UI development. I'm a backend programmer and ETL and Data Science are my most accomplished skillset in terms of what I *can* do. But often, you'll want a 'full-stack developer' in the classical sense too. I have stylistic Javascript and ecosystem style and functional OOP style knowledge. I can write CSS/HTMX, and I can build web apps in any language. I'm very partial to having a static asset stack where possible, Nginx, and I like Github Pages and Jekyll CMS. Most of my content is Markdown, plain-text, and org-mode when it's not software projects.
  

  
  With over five years in industry bioinformatics teams I have honed the essential professional skills of time-management, public speaking, and the technical accumen needed to deliver value for your team.
  

  * Knowledge:

<!--  With deep and well-rounded expertise in molecular biology, genetics, data analytics, and statistics, I have affected several teams and their publications, screening methods, and more. My team oriented style begins with a fundamentals first approach to discussing expertimental or analytical concerns with the Principal Investigator. When artifacts or biophysical aspects of assay design require special treatment, my backgorund in chemistry, biology, mathematics, and data analytics allows me to quickly integrate suggestions from team members into the analytical process, such as filtering bad leads or integrating additional knowledge for specific cases within large datasets. -->


  Committed to continuous learning and staying updated with the latest advancements in these fields, my breadth of knowledge allows me to model treatment conditions and their effects regardless of your experimental design. I write well and clear in technical and team messaging. 

>  Hi, I'm Matt and I'm both a wet-bench professional and software developer. 


## Background

> Univ. of Delaware B.Sc. Biochemistry and M.Sc. Bioinformatics and Computational Biology (3.96 GPA)

### Undergraduate program

I developed laboratory fundamentals as a wet-bench laboratory biochemist and molecular biologist. I learned aseptic culture, some immunological, and fluorescent assay techniques, and RT-PCR in a translational cancer laboratory. The UD chemistry department emphasized other laboratory fundamentals such as HPLC, GC-MS, spectrometry, and organic synthesis.

Subects studied include organic, phyical and biological chemistries, molecular biology, genetics, physiology, microbiology, virology, physics, linear algebra, probability, calc I/II/III and more.

### Graduate program

My thesis was hybrid wet-lab and dry-lab 2-year project developing the first strand-specific Illumina NGS map of the *C. acetobutylicum* transcriptome during solventogensis and sporulation stresses. Owing of course the help of my advisors Drs. Papoutsakis and Venkataramanan, I investigated the whole-transcriptome response of a biofuel-producing bacterium to the biophysical stresses of solvent/biofuel accumulation and associated stress-response adaptations in anaerobic fermentation conditions. I formulated a fixed media, prepared the bioreactors and innocula, sampled across time points for small molecule characterization by HPLC, RNA-quality investigation, subsequent PCR-mediated Illumina HiSeq 2500 library prepration, and delivered my samples to the Delaware Biotech. Institute core for sequencing. 

Afterwards, I developed both genome browser and interactive volcano-plot visualization web-applications for assessing differential expression of transcripts across the time-stress design.

Subjects included in my graduate program include metabolic engineering, bioinformatics, computational biology, database systems, algorithm topics, sequence alignment, Illumina NGS, biostatistics, Linux HPC training, web application development, Perl, Python and bourne shell (bash) programming. 


## Experience

### Wet-bench Molecular Biology and Biochemistry

I have worked in a variety of laboratory settings including analytical chemistry, organic synthesis, molecular biology and disease phsyiology research, translational oncology labs with patient tissue samples, extreme anaerobic and BSL-2 microbiolgy aseptic culture, and much more.

I am qualified to work with RT-PCR, immunological assays, immunofluorescence, Western blots, HPLC, GC/LC-MS, UV-VIS and more.

I keep a good laboratory notebook, I can rerun my samples independently if quality is off (e.g. RIN score), and I deeply enjoy performing molecular assays at the bench, whether its microbiology, human or mammalian pathology and physiology, or othewise.


### Computational

I primarily work in GNU/Linux environments on biological sequence analysis problems. I have many years of experience writing Python, Perl, R, Javascript, HTML/CSS, SQL, bash, and more recently, a little bit of Rust. 

My projects have included technologies such as kubernetes code, Dockerized workflows, shell scripts for HPC environments and grid engines like UGE/SGE and SLURM. I can build MVC/MVVC web applications and I studied database systems (SQL) at the graduate level.

I specialize in writing command-line interface (CLI) tools including languages like Python, Perl, R, NodeJS, and more recently, some Rust. Nvidia CUDA GPU programming and distributed environments are an additional strength for applications requiring excessive parallelism. 

Also, my experience with web applications, Javascript/HTML/CSS, front-end frameworks (mostly Jquery, Jquery-UI, D3.js, and SVG), REST-API design, and database architecture lets me create strong tools for developers and scientists alike.


Specific research experience stories are given below.

- Developed robust backend systems using Python, NodeJS, and R to streamline data processing workflows.
- Demonstrated excellence in laboratory skillsets with replicate correlation coefficients > 95% on Illumina HiSeq 2500 RNA-Seq libraries.
- Implemented cloud infrastructure on AWS, leveraging services such as EC2, S3, and Lambda for scalable and cost-effective solutions.
- Containerized applications and workflows using Docker and Kubernetes (k8s), ensuring consistency across development, testing, and production environments.
- Utilized Linux command line tools, including bash, grep, and sed, for data manipulation and system administration tasks.
- Designed and implemented interactive data visualizations using D3.js, enhancing data-driven decision-making processes.
- Scripted automation tasks in Perl and Python to optimize repetitive processes, reducing manual effort by 40%.
- Managed document preparation and typesetting using LaTeX for high-quality technical documentation.




## Testimonials

When asked for references, I've received the following from former managers.

Anonymous

> Whatever you need, big guy.

Anonymous, former boss and mentor

> Of course. Any time. You don't even need to ask.


