---
permalink: /
title: "5+ yrs. professional"
subtitle: Matt Ralston - results-driven Research Associate
author_profile: true
hero_image: "mattralston.png"
redirect_from: 
  - /about/
  - /about.html
---

## Western/Northern blot, RT-PCR, GC-MS, HPLC, ligation, restriction enzymes, ELISA, IHC/IF, FACS-Aria, Illumina library preparation. You name it!

## Bioinformatics specialist

Hello! My name is Matt and I'm a bench biologist and bioinformatics programmer. Welcome to my website! You can quickly get to where you're going below, or if you're an industry talent specialist or technical manager, I encourage you to read the 'About Me' section.

> "The scientific method is like a bicycle: you can go somewhere on it if you want to, but it's not really the sort of thing you can just play around with." - Terry Pratchett

## Sitemap

* [Portfolio](/portfolio)
* [Blog](/blog)
* [GitHub](https://github.com/MatthewRalston)
* [Resume/C.V.](/cv)
* [Publications](https://scholar.google.com/citations?hl=en&pli=1&user=86Lxg-4AAAAJ)

## Skills

## About Me

> Target audience: scientists, engineers, managers, and academic partners



Seeking a dynamic addition to your team? I bring 5 years of industry biopharmaceutical R&D team experience coupled with a decade of experience in wet-bench biology, Linux, and data analysis with Python/Perl/R. Let's elevate your projects together!

<!-- Are you looking for a new hire or academic collaborator with 5+ years of industry experience, 10+ years of bioinformatics, Linux, and Python/Perl/R coding expertise? -->

> Does culture, communication, productivity, and passion guide your decision in hiring new talent?

> Require a communicator adept at bridging diverse disciplines? I navigate the vibrant tapestry of wet-bench laboratory scientists, analysts, and managerial professionals alike. I listen to as many voices as I can, and I share my experiences with others; it's that good vibration when we share what is mutually understood and valued. Let's foster collaboration in your environment, soon!

<!-- Do you need someone with the communication style needed for a melting-pot of scientists, engineers, and managerial professionals to be heard, understood, and valued? -->

Effective communication **begins** with the "why?" and "how?" of a scientific question or modeling method. However, our teams flourish most when our individual communication style's lead everyone to the team's consensus while still allowing exploration of alternative ideas. My written style leads to "why don't we schedule you again this quarter?" instead of  "why am I in this meeting..."

<!-- I am an *expert* at modeling, data-processing, and visualization techniques such as *R*, *Cython*/*Python*, *Javascript*, *Linux*/*bash*, *Docker*, *AWS*/cloud and more. It's more than just websites, APIs, numbers and figures here; my strong fundamentals and data-driven design strategy have already been used to affect diverse goals in industry teams: hypothesis generation, key-questions, application or integration needs, and more, such that results are easily reproducibile, documented, and clearly written. -->

But I'm a bench scientist too! Apart from collaborating in group standups, breakout meetings, or quarterly huddles, I work independently and in teams to achieve progress for my managers yearly and quarterly performance and cultural endpoints. 


  * Culture:
  
  With some recommendations from industry managers, I've delivered value on projects within integrated teams of laboratory scientists, programmers, engineers, and managers on projects in wet-bench biology, bioinformatic and cheminformatic research areas. I take statistics, data, and modeling efforts seriously and I encourage you to check out the [testimonials](#testimonials) section and perhaps my modest GitHub [portfolio](/portfolio).
  
  * Communication:

  With proficiencies in technical writing, science, engineering, and stats/probability/mathematics, I've delivered value and knowledge to others without losing touch. Your high-value team demands written style and know-how in publications, meetings, and documentation. Clear and compelling written style is essential to todays business successes, tomorrows publications, and quarterly progress. Check out the [publications](/publications) or [testimonials](#testimonials) sections for more reasons why I'm the better fit for your business needs.

  * Productivity:
  
  As computational professionals know, there is more to productivity than LOC or turn-around time. Technical debt is *the* issue facing today's scientific and engineering teams. Specifically, how do we make deliverables correct and efficient, but also maintainable and compatible with adjacent apps, services and platforms? There's more to it than turn-around time; it's about integration and availability across sites, platforms, and teams.
  
  With over five years in industry bioinformatics teams I have honed the essential professional skills of time-management, public speaking, and the technical accumen needed to deliver value for your team.
  
  * Passion:
  
  I get genuine satisfaction from writing open-source code, algorithms, and pipelines. Scripts, APIs, websites, and databases are rapid turn-around deliverables in my perspective. Possessing a well-rounded STEM background, I bring expertise in diverse areas such as biological networks (metabolomics, transcriptomics, and genomics), mathematical and statistical theory, and nuanced experimental considerations including the biophysical origins of artifacts. Furthermore, I excel in modeling error, probability assessment, and programming design patterns. If you seek more than just a proficient coder, I strongly encourage you to delve deeper into my profile to discover the breadth of my capabilities.
  
  <!-- I have a balanced STEM background that includes biological networks (metabolomics, transcriptomics, and some human and non-human genomics), math/stats theory, even nuanced experimental concerns like the biophysical origin of artifacts, as well as modeling error, probability, and programming design patterns. If you want more than a coder, then I must passionately emphasize you read on to find out more. -->

  * Knowledge:

  With deep and well-rounded expertise in molecular biology, genetics, data analytics, and statistics, I have affected several teams and their publications, screening methods, and more. My team oriented style begins with a fundamentals first approach to discussing expertimental or analytical concerns with the Principal Investigator. When artifacts or biophysical aspects of assay design require special treatment, my backgorund in chemistry, biology, mathematics, and data analytics allows me to quickly integrate suggestions from team members into the analytical process, such as filtering bad leads or integrating additional knowledge for specific cases within large datasets.

  Committed to continuous learning and staying updated with the latest advancements in these fields, my breadth of knowledge allows me to model treatment conditions and their effects regardless of your experimental design. My data driven approach elucidates relationships, outlier measurements, response curves, and of course separates the effects in various assay designs. 

>  I'm more than just a wet-bench professional turned programmer. I'm your team's next valued member.


## Background

> Univ. of Delaware B.Sc. Biochemistry and M.Sc. Bioinformatics and Computational Biology (3.96 GPA)

### Undergraduate program

I developed laboratory fundamentals as a wet-bench laboratory biochemist and molecular biologist. I learned aseptic culture, some immunological, and fluorescent assay techniques, and RT-PCR in a translational cancer laboratory. The UD chemistry department emphasized other laboratory fundamentals such as HPLC, GC-MS, spectrometry, and organic synthesis.

Subects studied include organic, phyical and biological chemistries, molecular biology, genetics, physiology, microbiology, virology, physics, linear algebra, probability, calc I/II/III and more.

### Graduate program

My thesis was hybrid wet-lab and dry-lab 2-year project developing the first strand-specific Illumina NGS map of the *C. acetobutylicum* transcriptome during solventogensis and sporulation stresses. Owing of course the help of my advisors Drs. Papoutsakis and Venkataramanan, I investigated the whole-transcriptome response of a biofuel-producing bacterium to the biophysical stresses of solvent/biofuel accumulation and associated stress-response adaptations in anaerobic fermentation conditions. I formulated a fixed media, prepared the bioreactors and innocula, sampled across time points for small molecule characterization by HPLC, RNA-quality investigation, subsequent PCR-mediated Illumina HiSeq 2500 library prepration, and delivered my samples to the Delaware Biotech. Institute core for sequencing. 

Afterwards, I developed both genome browser and interactive volcano-plot visualization web-applications for assessing differential expression of transcripts across the time-stress design.

Subjects included in my graduate program include metabolic engineering, bioinformatics, computational biology, database systems, algorithm topics, sequence alignment, Illumina NGS, biostatistics, Linux HPC training, web application development, Perl, Python and bourne shell (bash) programming. 


## Experience

### Wet-bench Molecular Biology and Biochemistry

I have worked in a variety of laboratory settings including analytical chemistry, organic synthesis, molecular biology and disease phsyiology research, translational oncology labs with patient tissue samples, extreme anaerobic and BSL-2 microbiolgy aseptic culture, and much more.

I am qualified to work with RT-PCR, immunological assays, immunofluorescence, Western blots, HPLC, GC/LC-MS, UV-VIS and more.

I keep a good laboratory notebook, I can rerun my samples independently if quality is off (e.g. RIN score), and I deeply enjoy performing molecular assays at the bench, whether its microbiology, human or mammalian pathology and physiology, or othewise.


### Computational

I primarily work in GNU/Linux environments on biological sequence analysis problems. I have many years of experience writing Python, Perl, R, Javascript, HTML/CSS, SQL, bash, and more recently, a little bit of Rust. 

My projects have included technologies such as kubernetes code, Dockerized workflows, shell scripts for HPC environments and grid engines like UGE/SGE and SLURM. I can build MVC/MVVC web applications and I studied database systems (SQL) at the graduate level.

I specialize in writing command-line interface (CLI) tools including languages like Python, Perl, R, NodeJS, and more recently, some Rust. Nvidia CUDA GPU programming and distributed environments are an additional strength for applications requiring excessive parallelism. 

Also, my experience with web applications, Javascript/HTML/CSS, front-end frameworks (mostly Jquery, Jquery-UI, D3.js, and SVG), REST-API design, and database architecture lets me create strong tools for developers and scientists alike.


Specific research experience stories are given below.

- Developed robust backend systems using Python, NodeJS, and R to streamline data processing workflows.
- Demonstrated excellence in laboratory skillsets with replicate correlation coefficients > 95% on Illumina HiSeq 2500 RNA-Seq libraries.
- Implemented cloud infrastructure on AWS, leveraging services such as EC2, S3, and Lambda for scalable and cost-effective solutions.
- Containerized applications and workflows using Docker and Kubernetes (k8s), ensuring consistency across development, testing, and production environments.
- Utilized Linux command line tools, including bash, grep, and sed, for data manipulation and system administration tasks.
- Designed and implemented interactive data visualizations using D3.js, enhancing data-driven decision-making processes.
- Scripted automation tasks in Perl and Python to optimize repetitive processes, reducing manual effort by 40%.
- Managed document preparation and typesetting using LaTeX for high-quality technical documentation.




## Testimonials

When asked for references, I've received the following from former managers.

Anonymous

> Whatever you need, big guy.

Anonymous, former boss and mentor

> Of course. Any time. You don't even need to ask.



<!--
# Foobarbaz

This is the front page of a website that is powered by the [academicpages template](https://github.com/academicpages/academicpages.github.io) and hosted on GitHub pages. [GitHub pages](https://pages.github.com) is a free service in which websites are built and hosted from code and data stored in a GitHub repository, automatically updating when a new commit is made to the respository. This template was forked from the [Minimal Mistakes Jekyll Theme](https://mmistakes.github.io/minimal-mistakes/) created by Michael Rose, and then extended to support the kinds of content that academics have: publications, talks, teaching, a portfolio, blog posts, and a dynamically-generated CV. You can fork [this repository](https://github.com/academicpages/academicpages.github.io) right now, modify the configuration and markdown files, add your own PDFs and other content, and have your own site for free, with no ads! An older version of this template powers my own personal website at [stuartgeiger.com](http://stuartgeiger.com), which uses [this Github repository](https://github.com/staeiou/staeiou.github.io).

A data-driven personal website
======
Like many other Jekyll-based GitHub Pages templates, academicpages makes you separate the website's content from its form. The content & metadata of your website are in structured markdown files, while various other files constitute the theme, specifying how to transform that content & metadata into HTML pages. You keep these various markdown (.md), YAML (.yml), HTML, and CSS files in a public GitHub repository. Each time you commit and push an update to the repository, the [GitHub pages](https://pages.github.com/) service creates static HTML pages based on these files, which are hosted on GitHub's servers free of charge.

Many of the features of dynamic content management systems (like Wordpress) can be achieved in this fashion, using a fraction of the computational resources and with far less vulnerability to hacking and DDoSing. You can also modify the theme to your heart's content without touching the content of your site. If you get to a point where you've broken something in Jekyll/HTML/CSS beyond repair, your markdown files describing your talks, publications, etc. are safe. You can rollback the changes or even delete the repository and start over -- just be sure to save the markdown files! Finally, you can also write scripts that process the structured data on the site, such as [this one](https://github.com/academicpages/academicpages.github.io/blob/master/talkmap.ipynb) that analyzes metadata in pages about talks to display [a map of every location you've given a talk](https://academicpages.github.io/talkmap.html).

Getting started
======
1. Register a GitHub account if you don't have one and confirm your e-mail (required!)
1. Fork [this repository](https://github.com/academicpages/academicpages.github.io) by clicking the "fork" button in the top right. 
1. Go to the repository's settings (rightmost item in the tabs that start with "Code", should be below "Unwatch"). Rename the repository "[your GitHub username].github.io", which will also be your website's URL.
1. Set site-wide configuration and create content & metadata (see below -- also see [this set of diffs](http://archive.is/3TPas) showing what files were changed to set up [an example site](https://getorg-testacct.github.io) for a user with the username "getorg-testacct")
1. Upload any files (like PDFs, .zip files, etc.) to the files/ directory. They will appear at https://[your GitHub username].github.io/files/example.pdf.  
1. Check status by going to the repository settings, in the "GitHub pages" section

Site-wide configuration
------
The main configuration file for the site is in the base directory in [_config.yml](https://github.com/academicpages/academicpages.github.io/blob/master/_config.yml), which defines the content in the sidebars and other site-wide features. You will need to replace the default variables with ones about yourself and your site's github repository. The configuration file for the top menu is in [_data/navigation.yml](https://github.com/academicpages/academicpages.github.io/blob/master/_data/navigation.yml). For example, if you don't have a portfolio or blog posts, you can remove those items from that navigation.yml file to remove them from the header. 

Create content & metadata
------
For site content, there is one markdown file for each type of content, which are stored in directories like _publications, _talks, _posts, _teaching, or _pages. For example, each talk is a markdown file in the [_talks directory](https://github.com/academicpages/academicpages.github.io/tree/master/_talks). At the top of each markdown file is structured data in YAML about the talk, which the theme will parse to do lots of cool stuff. The same structured data about a talk is used to generate the list of talks on the [Talks page](https://academicpages.github.io/talks), each [individual page](https://academicpages.github.io/talks/2012-03-01-talk-1) for specific talks, the talks section for the [CV page](https://academicpages.github.io/cv), and the [map of places you've given a talk](https://academicpages.github.io/talkmap.html) (if you run this [python file](https://github.com/academicpages/academicpages.github.io/blob/master/talkmap.py) or [Jupyter notebook](https://github.com/academicpages/academicpages.github.io/blob/master/talkmap.ipynb), which creates the HTML for the map based on the contents of the _talks directory).

**Markdown generator**

I have also created [a set of Jupyter notebooks](https://github.com/academicpages/academicpages.github.io/tree/master/markdown_generator
) that converts a CSV containing structured data about talks or presentations into individual markdown files that will be properly formatted for the academicpages template. The sample CSVs in that directory are the ones I used to create my own personal website at stuartgeiger.com. My usual workflow is that I keep a spreadsheet of my publications and talks, then run the code in these notebooks to generate the markdown files, then commit and push them to the GitHub repository.

How to edit your site's GitHub repository
------
Many people use a git client to create files on their local computer and then push them to GitHub's servers. If you are not familiar with git, you can directly edit these configuration and markdown files directly in the github.com interface. Navigate to a file (like [this one](https://github.com/academicpages/academicpages.github.io/blob/master/_talks/2012-03-01-talk-1.md) and click the pencil icon in the top right of the content preview (to the right of the "Raw | Blame | History" buttons). You can delete a file by clicking the trashcan icon to the right of the pencil icon. You can also create new files or upload files by navigating to a directory and clicking the "Create new file" or "Upload files" buttons. 

Example: editing a markdown file for a talk
![Editing a markdown file for a talk](/images/editing-talk.png)

For more info
------
More info about configuring academicpages can be found in [the guide](https://academicpages.github.io/markdown/). The [guides for the Minimal Mistakes theme](https://mmistakes.github.io/minimal-mistakes/docs/configuration/) (which this theme was forked from) might also be helpful.




-->
