---
permalink: /about
title: "5+ yrs. professional"
subtitle: Matt Ralston - results-driven Research Associate
author_profile: true
hero_image: "mattralston.png"
toc: true
---



> "The scientific method is like a bicycle: you can go somewhere on it if you want to, but it's not really the sort of thing you can just play around with." - Terry Pratchett



# About Me

> Target audience: scientists, engineers, managers, and academic partners


Seeking a dynamic addition to your team? I bring 5 years of industry biopharmaceutical R&D team experience coupled with a decade of experience in wet-bench biology, Linux, and data analysis with Python/Perl/R. Let's **elevate** your projects together!


## Navigate

* [Industry experience](#industry-experience)
* - the auspicious [Wet lab](#wet-bench-molecular-biology-and-biochemistry)
* - the rumored [Dry lab](#science-and-computation-meets-software-development-and-agile-teams)
* [Academic background](#training)
* [Skills](#skills)
* - the glamorous - [Wet laboratory skillsets](#wet-laboratory-skills)
* - the powerhouse - [devops, ML, AI, and engineering](#engineering-skills)
* [Who is Matt Ralston?](#matt-in-a-nutshell)
* - [Team skills](#team-skills) - Culture, communication, teamwork, and professional empathy
* [Contact](#contact)

> Do you consider cultural values, communication style, productivity, and dedication in your decision to hire new talent? Do you require a professional adept at communicating in-between and bridging diverse disciplines? 

Do you need to know something more specific about my [software projects](/software), [industry experience](/cv), or my [laboratory training](/biosciences)? **Use the links here or on the [main page/sitemap](/) to find out more.**




# Industry experience

### Wet-bench Molecular Biology and Biochemistry

I have worked in a variety of laboratory settings including analytical chemistry, organic synthesis, molecular biology and disease physiology research, translational oncology labs with patient tissue samples, extreme anaerobic and BSL-2 microbiolgy aseptic culture, and much more. In addition to laminar-flow hoods and GLP environments, I am familiar with a wide range of instrumentation and bioassay designs. I have worked with qRT-PCR, immunological assays, immunofluorescence, Northern/Western blots, HPLC, GC/LC-MS, UV-VIS and more.

I keep a good laboratory notebook, I can run experiments independently if sample quality is off (e.g. RIN score), and I deeply enjoy performing molecular assays at the bench, whether its microbiology, human or mammalian pathology and physiology, or otherwise.

If you're looking for someone who can aliquot 1uL to instrument tolerance, or who can run a multi-factor experiment or bioassay from innoculation through sample preparation and instrumental analysis, I'm your guy.


### Science and computation meets software development and agile teams

I primarily work in GNU/Linux environments on biological sequence analysis problems. I have many years of experience writing Python, Perl, R, Javascript, HTML/CSS, SQL, bash, and more recently, a little bit of Rust. 

My projects have included technologies such as kubernetes code, Dockerized workflows, shell scripts for HPC environments and grid engines like UGE/SGE and SLURM. I can build MVC/MVVC web applications and I studied database systems (SQL) at the graduate level.

I specialize in writing command-line interface (CLI) tools including languages like Python, Perl, R, NodeJS, and more recently, some Rust. Nvidia CUDA GPU programming and distributed environments are an additional strength for applications requiring excessive parallelism. 

Also, my experience with web applications, Javascript/HTML/CSS, front-end frameworks (mostly Jquery, Jquery-UI, D3.js, and SVG), REST-API design, and database architecture lets me create strong tools for developers and scientists alike.


Specific research experience stories are given below.

- Developed robust backend systems using Python, NodeJS, and R to streamline data processing workflows.
- Demonstrated excellence in laboratory skillsets with replicate correlation coefficients > 95% on Illumina HiSeq 2500 RNA-Seq libraries.
- Implemented cloud infrastructure on AWS, leveraging services such as EC2, S3, and Lambda for scalable and cost-effective solutions.
- Containerized applications and workflows using Docker and Kubernetes (k8s), ensuring consistency across development, testing, and production environments.
- Utilized Linux command line tools, including bash, grep, and sed, for data manipulation and system administration tasks.
- Designed and implemented interactive data visualizations using D3.js, enhancing data-driven decision-making processes.
- Scripted automation tasks in Perl and Python to optimize repetitive processes, reducing manual effort by 40%.
- Managed document preparation and typesetting using LaTeX for high-quality technical documentation.


# Training

> Univ. of Delaware B.Sc. Biochemistry and M.Sc. Bioinformatics and Computational Biology (3.96 GPA)

### Undergraduate program

I developed laboratory fundamentals as a wet-bench laboratory biochemist and molecular biologist. I learned aseptic culture, some immunological, and fluorescent assay techniques, and RT-PCR in a translational cancer laboratory. The UD chemistry department emphasized other laboratory fundamentals such as HPLC, GC-MS, spectrometry, and organic synthesis.

Subects studied include organic, phyical and biological chemistries, molecular biology, genetics, physiology, microbiology, virology, physics, linear algebra, probability, calc I/II/III and more.

### Graduate program

My thesis was hybrid wet-lab and dry-lab 2-year project developing the first strand-specific Illumina NGS map of the *C. acetobutylicum* transcriptome during solventogensis and sporulation stresses. Owing of course the help of my advisors Drs. Papoutsakis and Venkataramanan, I investigated the whole-transcriptome response of a biofuel-producing bacterium to the biophysical stresses of solvent/biofuel accumulation and associated stress-response adaptations in anaerobic fermentation conditions. I formulated a fixed media, prepared the bioreactors and innocula, sampled across time points for small molecule characterization by HPLC, RNA-quality investigation, subsequent PCR-mediated Illumina HiSeq 2500 library prepration, and delivered my samples to the Delaware Biotech. Institute core for sequencing. 

Afterwards, I developed both genome browser and interactive volcano-plot visualization web-applications for assessing differential expression of transcripts across the time-stress design.

Subjects included in my graduate program include metabolic engineering, bioinformatics, computational biology, database systems, algorithm topics, sequence alignment, Illumina NGS, biostatistics, Linux HPC training, web application development, Perl, Python and bourne shell (bash) programming. 

# Skills

My skillsets fall into four categories:

## Wet laboratory skills

* Laboratory sciences major (biology and chemistry specialization BSc. Biochemistry, UD '12)

I have core competencies in laboratory skills, soft skills related to team organization and responsibility and relevant course work covering [molecular biology, human and mammalian genomics, mammalian physiology, microorganism genomics (bacterial, viral) metagenomics and phylogenomics, biofuels and cell culture bioreactor,] process optimization, [novel methodologies and protocols, Illumina RNA-seq library preparation, qRT-PCR, RT-PCR, PAGE: Western blot, Northern blot, ELISA, HPLC, nucleic acid extraction, enzymatic treatment], FACS, [light microscopy, fluorescence microscopy, immunohistochemistry, immunofluorescence, gene knockdown, cloning,] bioassay development, and additionally [media formulation, contaminant testing and detection], [technical and software expertise], [GLP, ELN], and additional undergraduate lab and bioscience, [natural sciences, undergraduate mathematics, probability, statistics, molecular diversity], [data standards and reporting, technical writing, trends and conventions in biodata archiving], 

Spectrometry: UV-Vis, Fluorometry, Nanodrop, MS methods... some training related to tandem MS and flux modeling.

Cell culture: Human/mammalian cell culture, anaerobic bacterial cell culture, fermentation, 

## Engineering skills

* Computer science major (masters of science in Comptuer Science, concentration in bioinformatics, 3.96 GPA, UD '15)

* Bioinformatics

Sequence alignment: Hisat2, Bowtie2, Tophat, BWA, Blast, Blat, ClustalO, HMMs

Assembly: Cufflinks, Trinity, Velvet

Illumina data processing: Fastqc, Samtools, Picard, Bedtools

Miscellaneous: Tuxedo suite, `samtools`, DEseq, limma, Circos, PFAM, David

* Software development

Cloud and containers: Amazon Web Services (AWS), Docker, OCI, k8s

Programming: Python, R, Bash, Javascript/NodeJS, Rust, Emacs, D3.js, Julia, Perl, awk, grep/sed, LaTeX, Matlab, HTML/CSS, MySQL, Oracle, PostgreSQL, miniconda

Data Management: Oracle SQL, PostgreSQL, rdflib, snapshotting

Cloud Computing: S3, Elastic Compute (EC2), CodeBuild, Elastic Container Repository (ECR), Elastic Container Service (ECS), Docker, IAM, Azure/GCP/AWS k8s

Networking: Slurm, Sun Gride Engine (SGE/UGE), firewalld, iptables, tmux, nixos

Systems Administration: systemd, Docker, Ansible/chef, tmux, Nginx, Apache, web servers, Sun Grid Engine (SGE/UGE), bash, parallel

* Statistics and data science

Statistics: Hypothesis testing, ANOVA/regression, distribution fitting, R/Rstudio/Rshiny, Bioconductor, discrete/continuous probability, multivariate statistical analysis, regression, least-squares estimation, PCA, clustering, classification, normalization, regularization, variance reduction, naive Bayes, random forest, XGBoost, 

Dimensionality reduction: PCA/SVD, Uniform Manifold Approximation and Projection (UMAP), t-SNE, canonical correlation analysis, 

Optimizers: Gradient descent, simplex, particle swarm, simulated annealing, self-organizing maps, others.

Kernels: kernel trick, linear programming, matrix math, linear algebra, abstract algebra, discrete math

Maching learning: perceptron, SVM, PCA, UMAP, t-SNE, DNN, RNN, recurrence relations, linear regression and formal OLS-family assumptions and caveats, SAS programming, R programming, Python programming

Data Science: Python programming, numpy, pandas, staticly-typed compiled languages (`rustc`/`cargo`, `zig`, Typescript, Scala/Java/Clojure, C extensions in Python and R), RAII, CUDA, Cython, anaconda, apline linux, Docker, k8s




# Team skills

* Culture:
  
With good recommendations from industry managers, I've delivered value on projects within integrated teams of laboratory scientists, programmers, engineers, and managers on projects in wet-bench biology, bioinformatic and cheminformatic research areas. I take statistics, data, and modeling efforts seriously and I encourage you to check out the my [/software](/software) page, the [/research](/research) page, my life sciences message at [/biosciences](/biosciences), my inactive profile on [LinkedIn](https://linkedin.com/in/matthewralston), and my manager [testimonials](#testimonials) section and perhaps my modest publicly facing [GitHub](https://github.com/MatthewRalston) and [/portfolio](/portfolio) page. I hope that you'll see I have some familiarity with publishing and I write usable and effective documentation, literate programming, and reproducibilty reports and analyses. I boast 3-5 modest open source genomics projects concerning sequence alignment conceptural understanding, molecular species and diversity, correlation analyses, k-mer and minimizer expertise, Illumina data profiling, and I am not too opinionated about preferences execpting my engineering training, my scientific and code development process, and my R&D process.

  
* Communication:

With proficiencies in technical writing, science, engineering, and stats/probability/mathematics, I've delivered value and knowledge to others without losing touch. Your high-value team demands written style and know-how in publications, meetings, and documentation. Clear and compelling written style is essential to today's business successes, tomorrow's publications, and our quarterly progress. Check out the [publications](/publications) or [testimonials](#testimonials) sections for more reasons why I'm a just **above an ordinarily spectacular fit** for your business needs.


{% include team_skills.md %}

# Contact

## Do you want to connect?

If you have a research question or you would like to discuss wet-bench, software, or bioinformatics please let me know. 

If you are a recruiter please provide a location, time-frame to hire, and a full job description as plain-text in an email to my [gmail](mailto:mralston.development@gmail.com).

If you are an interviewer and would like to schedule a phone or zoom interview, please copy and paste the following text into your email.

```
name: Jane Doe
email: jane.doe@example.com
company: Example dot com
department: Data engineering
subject: I'd like to contact you about the following opportunity
date: Date (- of opportunity)
time: time of interview
description: A description of the position or opportunity you'd like to disclose to me.
question: Key questions I should address in my response 
kpi: What are your indicators of whether or not this is successful, like a successful interview shedule by a certain date. A fee range you'd like to remain within. What makes our relationship successful and how may I discuss details more
```

# [Contact](/contact)

{% include contact.md %}

----------------------------------------------------------------------




## Testimonials

When asked for references, I've received the following from former managers.

Anonymous

> Whatever you need, big guy.

Anonymous, former boss and mentor

> Of course. Any time. You don't even need to ask.


