---
permalink: /about
title: "5+ yrs. professional"
author: ralston
date: 2025-03-15 03:44:00 +0400
layout: post
categories: [About]
tags: [About]
render_with_liquid: false
hero_image: "mattralston.png"
---



> "The scientific method is like a bicycle: you can go somewhere on it if you want to, but it's not really the sort of thing you can just play around with." - Terry Pratchett



# About Me

> Target audience: scientists, engineers, managers, and academic partners


Seeking a dynamic addition to your team? I bring 5 years of industry biopharmaceutical R&D team experience coupled with a decade of experience in wet-bench biology, Linux, and data analysis with Python/Perl/R/Rust. Let's **elevate** your projects together!



> Do you consider *cultural values, communication style, productivity, and dedication* in your decision to hire new talent? Do you require a *professional adept at communicating with engineers and scientists from diverse disciplines*? 

Do you need to know something more specific about my [laboratory](/biosciences) experience, [software](/software) projects, [data_science](/data_science) experience, [industry experience](/cv), or [coding](/about#skills) proficiencies? **Use the links here or on the [main page/sitemap](/) to find out more.**




# Industry experience


### Wet-bench Molecular Biology and Biochemistry

> 5+ years of published laboratory biochemistry and molecular biology expertise

I have worked in a variety of laboratory settings including analytical chemistry, organic synthesis, molecular biology and disease physiology research, translational oncology labs with patient tissue samples, extreme anaerobic and BSL-2 microbiolgy aseptic culture, and much more. In addition to laminar-flow hoods and GLP environments, I am familiar with a wide range of instrumentation and bioassay designs. I have worked with qRT-PCR, immunological assays, immunofluorescence, Northern/Western blots, HPLC, GC/LC-MS, UV-VIS and more. Please see the [skills](#skills) section for specific details and a list of known lab protocols.

I keep a meticulous laboratory notebook. I can re-run samples if quality is off (e.g. RIN score). Also, I deeply enjoy performing molecular assays at the bench, whether its genomics, microbiology, human or mammalian disease pathology, genetics, and physiology. I am detail oriented and very organized and I focus on replicate quality in-between samples and I understand instrument limitations, positive controls, and core goals of hypothesis testing and data modeling.

If you're looking for a hybrid wet-bench and data scientist who can aliquot 1uL reagents to pipette instrument tolerance, or who can run a multi-factor experiment or bioassay from innoculation through sample preparation and instrumental analysis, I'm your guy.


### Agile scientific and cloud computing meets software development expertise

> 10+ years of software development experience

I primarily work in GNU/Linux and MacOS/OSX environments  on biological sequence analysis, data science, and **Illumina short-read WGS/WES/RNAseq** problems. I have 5 years of industry experience writing **Python, Perl, R, Javascript, HTML/CSS, SQL,** bash and recently, web applications (React.js, Leptos, WASM) and safe-memory GUIs/TUIs with **Rust**. 

My projects have included technologies such as kubernetes code, Dockerized workflows, shell scripts for HPC environments and grid engines like UGE/SGE and SLURM. I can build MVC/MVVC web applications and I studied database systems (SQL) at the graduate level.

I specialize in writing command-line interface (CLI) tools including languages like Python, Perl, R, NodeJS, and Rust. Nvidia CUDA GPU programming and distributed environments are an additional strength for applications requiring excessive parallelism. 

Also, my experience with web application development, relational databases (Microsoft SQL, PostgreSQL, SQLite3 and libSQL, Oracle SQL, Amazon RDS (Relational Database Service) and equivalent on commodity cloud vendors), Javascript/HTML/CSS, front-end frameworks (mostly Alpine.js, React, Jquery, Jquery-UI, D3.js, and SVG), REST-API design, and database normalizations and architecture lets me create strong tools for developers and scientists alike.


Specific stories and industry experiences are given below.

- Developed robust backend systems using Python, NodeJS, and R to streamline data processing workflows.
- Demonstrated excellence in laboratory skillsets with replicate correlation coefficients > 95% on Illumina HiSeq 2500 RNA-Seq libraries.
- Implemented cloud infrastructure on AWS, leveraging services such as EC2, S3, and Lambda for scalable and cost-effective solutions.
- Containerized applications and workflows using Docker and Kubernetes (k8s), ensuring consistency across development, testing, and production environments.
- Utilized Linux command line tools, including bash, grep, and sed, for data manipulation and system administration tasks.
- Designed and implemented interactive data visualizations using D3.js, enhancing data-driven decision-making processes.
- Scripted automation tasks in Perl and Python to optimize repetitive processes, reducing manual effort by 40%.
- Managed document preparation and typesetting using LaTeX for high-quality technical documentation.


# Training

> Univ. of Delaware B.Sc. Biochemistry (2012) and M.Sc. Bioinformatics and Computational Biology (3.96 GPA, 2014)

### Undergraduate program

I developed laboratory fundamentals as a wet-bench laboratory biochemist and molecular biologist. I learned aseptic culture, some immunological, and fluorescent assay techniques, and RT-PCR in a translational cancer laboratory. The UD chemistry department emphasized other laboratory fundamentals such as HPLC, GC-MS, spectrometry, and organic synthesis.

Subects studied include organic chemistry, physical chemistry and thermodynamics, and biochemistry, molecular biology, genetics, physiology, microbiology, virology, physics, linear algebra, probability, calc I/II/III and others.

### Graduate program

My thesis was hybrid wet-lab and dry-lab 2-year project developing the first strand-specific Illumina NGS map of the *C. acetobutylicum* transcriptome during solventogensis and sporulation stresses. Owing of course to the help of Drs. Papoutsakis and Venkataramanan, I investigated the whole-transcriptome response of a biofuel-producing bacterium to the biophysical stresses of solvent/biofuel accumulation and associated stress-response adaptations in anaerobic fermentation conditions. I formulated a fixed media, prepared the bioreactors and innocula, sampled across time points for small molecule characterization by HPLC, RNA-quality investigation, subsequent PCR-mediated Illumina HiSeq 2500 library prepration, and delivered my samples to the Delaware Biotech. Institute core for sequencing. 

Afterwards, I developed both genome browser and interactive volcano-plot visualization web-applications for assessing differential expression of transcripts across the time-stress design.

Subjects included in my graduate program include metabolic engineering, bioinformatics, biostatistics, computational biology, database systems, algorithm topics, sequence alignment, Illumina NGS, biostatistics, Linux HPC training, web application development, Perl, Python and bourne shell (bash) programming. 


# Skills

## Wet laboratory skills

* Laboratory sciences BSc. (Chemistry department bachelors degree with specialization: **BSc. Biochemistry,** University of Delaware, 2012)

I have demonstrated competencies in many laboratory methods as well as soft skills related to team organization, and I have graduate-level course work and **10 years of industry and academic experience** covering [molecular biology, **human and mammalian genomics and immortalized cell-line culture,** mammalian physiology, microorganism genomics (bacterial, viral) metagenomics and phylogenomics, biofuels and cell culture bioreactor,] process optimization, [novel methodologies and protocols, **Illumina RNA-seq library preparation, qRT-PCR, RT-PCR, PAGE (Western blot, Northern blot), ELISA, Chromatography (TLC, HPLC), Mass Spectrometry (GC-MS, MS-MS, LC-MS),** RNA quality analysis/RIN, Nanodrop, nucleic acid extraction, enzymatic treatment], Flow Cytometry/FACS-Aria, [light microscopy, fluorescence microscopy, immunohistochemistry (IHC), immunofluorescence (IF), gene knockdown, cloning,] bioassay development, and additionally [media formulation, contaminant testing and detection], [technical and software expertise], **[GLP, ELN],** and additional undergraduate lab and bioscience, [natural sciences, undergraduate mathematics, probability, statistics, molecular diversity], [data standards and reporting, **technical writing,** trends and conventions in biodata standards]. 

>NOTE: Skills listed above in brackets indicate genuine industry experiences and/or experience at a professional level.

## Engineering skills


* Computer Science MSc. (**MSc. in Comptuer Science**, concentration in bioinformatics, **3.96 GPA,** UD 2015)

In addition to my considerable training in analytical/physical/organic chemistries and molecular biology and genomics, I possess very strong skills in programming and data science. With a bioinformatics major in graduate school at Univ. of Delaware, I have formal training in advanced calculus, machine learning and statistics,  and software engineering. 

Additionally, I have practiced many skills in industry, leading efforts in Bristol-Myers Squibb company initiatives such as legacy software development and maintainance, reproducible research, dashboard development, software application development, database management and data version control, and programming language adoption.


### Software development

- Cloud and containers: **Amazon Web Services (AWS)**, **Docker** and related container technologies, `containerd`, OCI, **kubernetes**/k8s

- Programming: **Rust, Python/C-Python, R, R-Shiny, bioconductor and Rmarkdown/Quarto, `bash`, Javascript/NodeJS/Bun,** `emacs`, D3.js, Julia, **Perl, `awk`, `grep`, `sed`, LaTeX,** Matlab, **HTML5/CSS3,** SQLite3/libSQL, MySQL/MariaDB, **Oracle SQL, MS SQL Server, PostgreSQL,** miniconda

- [Relational Databases](https://en.wikipedia.org/wiki/Relational_database) and noSQL technologies: MongoDB, Microsoft SQL Server, Oracle SQL, MariaDB/MySQL, SQLite3/libSQL, PostgreSQL, SQLalchemy and Object-relational-mapping (ORM), Rust ORMs, `sqlx`, rdflib, snapshotting

- [Cloud Computing](https://en.wikipedia.org/wiki/Cloud_computing) and [Cloud Native](https://aws.amazon.com/what-is/cloud-native/): **GCP, Azure, AWS**, S3, Elastic Compute (EC2), CodeBuild/CodeDeploy, Elastic Container Repository (ECR), Elastic Container Service (ECS), Docker, IAM,  kubernetes/k8s

- Networking: Slurm, Sun Gride Engine (SGE/UGE), firewalld, iptables, `tmux`, nixos

- Systems Administration: `systemd`, Docker, Ansible/chef, `tmux`, **`nginx` and Apache/`httpd` web servers, Sun Grid Engine (SGE/UGE) and SLURM,** bash programming, parallel

### Statistics and data science

- Statistics: **p-values,** Student **t-test,** Chi-Square test, **Fisher's Exact Test**, Hypothesis testing, **ANOVA/GLM**/regression, linear algebra, distribution fitting, *R/Rstudio/Rshiny*, Bioconductor, discrete/continuous probability, multivariate statistical analysis, regression, least-squares estimation, PCA, clustering, classification, normalization, regularization, variance reduction, naive Bayes, random forest, XGBoost, 

- Dimensionality reduction: PCA/SVD, **Uniform Manifold Approximation and Projection (UMAP), t-SNE,** canonical correlation analysis, 

- Optimizers: Gradient descent, simplex, particle swarm, simulated annealing, self-organizing maps, others.

- Kernels: kernel trick, linear programming, matrix math, linear algebra, abstract algebra, discrete math

- Maching learning: perceptron, **SVM, PCA, UMAP, t-SNE, Deep Neural Networks, Recurrent Neural Networks, and Artificial Intelligence (AI)** recurrence relations, linear regression and formal OLS-family assumptions and caveats, **SAS programming, R programming, Python programming**

- Data Science: Python programming, `numpy`, `pandas`, **Jupyter notebooks, Markdown, LaTeX/Pandoc,** staticly-typed compiled languages (`rustc`/`cargo`, `zig`, Typescript, Scala/Java/Clojure, C extensions in Python and R), RAII, **CUDA**, Cython, anaconda, apline linux, Docker, kubernetes/k8s, 


### Bioinformatics

- Sequence alignment: Hisat2, Bowtie2, Tophat, BWA, Blast, Blat, ClustalO, Hidden Markov Models, E-value

- Assembly: Cufflinks, Trinity, Velvet

- Illumina data processing: Fastqc, Samtools, Picard, Bedtools

- Miscellaneous: Tuxedo suite, `samtools`, DEseq, limma, Circos, PFAM, David



## Team skills

* Culture:
  
With excellent recommendations from industry managers on the topics of productivity, code maintainance, and team behaviors, I've delivered value on projects within integrated teams of laboratory scientists, programmers, engineers, and managers on projects in wet-bench biology, bioinformatic and cheminformatic research areas. I take statistics, data, and modeling efforts seriously and I encourage you to check out the my [/software](/software) page, the [/research](/research) page, my life sciences message at [/biosciences](/biosciences), my profile on [LinkedIn](https://linkedin.com/in/matthewralston), and my manager [testimonials](#testimonials) section and perhaps my publicly facing [GitHub](https://github.com/MatthewRalston) and [/portfolio](/portfolio) pages. You'll also see I have expert familiarity with publishing first-author and collaborating author articles. I write usable and effective documentation, I am a guru with literate programming topics such as Rmarkdown, markdown, Pandoc, Infrastructure-as-code (IAC), and reproducibilty reports and analyses. I boast 3-5 modest open source genomics projects concerning sequence alignment conceptural understanding, molecular species and diversity, correlation analyses, k-mer and minimizer expertise, and Illumina sequence data profiling. I am enthusiastic about my software engineering, my scientific and code development, and my Discovery and R&D processes.

  
* Communication:

With proficiencies in technical writing, science, engineering, and stats/probability/mathematics, I've delivered value and knowledge to others without losing touch. Your high-value team demands written style and know-how in publications, meetings, and documentation. Clear and compelling written style is essential to today's business successes, tomorrow's publications, and your quarterly progress. Check out the [publications](/publications) or [testimonials](#testimonials) sections for more reasons why I'm a **above ordinary and spectacular fit** for your business needs.






## Who am I?

I'm Matt Ralston and I'm a data science, scientific computing, devops, and knowledge-economy professional. If you want to get to know me, please check out the various topics you can find throughout my website on the [/about](/about) me page.

### Matt in a nutshell

I work in groups of engineers, analysts, and wet-bench laboratory professionals, team leads, associate directors, and alike. I listen to as many voices as I can, and I share my experiences with others; I am detail oriented and I enjoy the problem solving process, and I can discuss your needs in a variety of ways going forwards.

* - has soft skills, style and brevity in communication
* - track-record and wide technical expertise in engineering, chemistry, biology and genomics, bioinformatics, and Illumina WGS/WES/RNAseq/metagenomics
* - knowledge of agile methodology
* - system design mindset: design, benchmarking and testing
* - cloud-native and Docker containers

### Go-to menu | statements considered harmful

* 
* - skip to [\[Communications and documentation\]](#communications-and-documentation)
* - skip to [\[Task and time management\]](#regarding-task-and-time-management)
* - skip to [\[server code vs UI/UX\]](#software-for-backend-and-server-code-vs-UI-and-UX)


## Life science specialty

And, so, I do wet-laboratory science as well. I'm effecient and effective and timely in the lab. Asepsis leads to less errors, and as a bench molecular biologist I'm familiar with the autoclave and where the imaging equipment is. I can perform single-gene targetted expression studies (sometimes referred to as qPCR or rt-PCR on mRNA molecules. I also work with Illumina lab reagent kit technologies). 

## Disclaimer and software licensing

I am certainly **not** an *expert* at modeling, data-processing, and visualization techniques such as *R*, *Cython*/*Python*, *Javascript*, *Linux*/*bash*, *Docker*, *AWS*/cloud and more. But It's more than just websites, APIs, numbers and figures here; my strong fundamentals and data-driven design strategy have been employted towares diverse goals in industry: hypothesis generation, key-questions and KPIs, application or integration needs, and more, such that my products are reproducibile, documented, and clearly written. I can adapt to your team.


I am *familiar* with open source standards, commercial licensing and procurement, software licenses, and EULAs.

And god willing I do multiplex in the lab and on the computer.

I have some conceptual familiarity with a lot of different methods from chemical analytical, bioassay, immunoassay, fluorimetry, MS, GC-MS, LC-MS, ELISA, and an intimate understanding of cell culture and laminar flow hoods. 

My last acura was an Illumina but my last published report was validations of Northerns and novel lnc-RNAs, antisense RNAs, RNAi, and Illumina transcriptomics with strand-specific protocols.


## Communications and Documentation

I work independently with consistent email and Slack communications, and project updates and logging via Version Control and `git`. My project management software is often plain text, also git-based, metrics, org-files for Pandoc compatibility, second-brain, markdown where possible, license documents, source files and configurations, Dockerfiles where possible, and I write LaTeX primarily in `.Rmd`.

Committed to continuous learning and staying updated with the latest advancements in these fields, my breadth of knowledge allows me to model, program, and problem solve in engineering teams. With my natural science knowledge I can ask intelligent questions about variables and covariates, treatment conditions, and their effects in your experimental design. I write clearly in technical and team messaging. 

I have stylistic Javascript and UI/UX ecosystem and preference set and functional and OOP style knowledge. I can write CSS/HTMX, and I can build web apps in any language. I'm very partial to having a static asset stack where possible, Nginx, and I like Github Pages and Jekyll CMS. Most of my content is Markdown, plain-text, and org-mode when it's not software projects.

That being said, most of my technical reporting is LaTeX, Pandoc, and Rmarkdown for scientific documentation.

I use Sphinx and Markdown for most simple code-and-a-story literate programming.

I am no stranger to Flask, FastAPI, Dask, OpenMPI, LAPACK, Tableau, Jupyter, and other standard app-and-reporting codebase trends and ease-of-use and lifetime centric development.

I've worked in Agile teams and I am familiar with scrum, sprints, stories and other development team concepts.

I've used Slack, Skype, and Atlassian distributed-team software and I am consistent on issues tracking, reporting, kanban, application lifetime management suggestions, and of course version control and repository feng-shui.




## GTD/Productivity mindset


Productivity is a modest topic for discussion and I find it enjoyable because you see others perspectives on engineering concerns, assumptions, testing needs, Blue-Green workflows, Red-team and Green-team development, and of course GTD methods when you're with new colleagues and their perspectives.

### Regarding task and time management

My workflow centers on the `README.md`, software usage patterns and documentation, Sphinx, Oxygen, and Pandoc documentation, literate programming, org-mode, and Emacs.

I use the zed editor, PyCharm, Emacs and vi, tmux, and the bash shell for my development environment and I focus heavily on commit messsages, gitflow, rebase/merge/squash source tree design, and I use plain text and org-mode to create and manage my TODO lists and organize information in a Pandoc compatible way.

My task and time management largely revolves in Atlassian and Github kanban and issue tracking systems. I also can use Emacs functionality to perform clock-ins on tasks and git to manage versions of documentation and code, in source trees, to perform time estimates and prototype cost estimation.

I am flexible regarding your task and time tracking and coding requirements speicified by your org chart and PM guidelines.


### Software for backend and server code vs UI and UX

I use Rmarkdown and Python for most of my direct data science stack. I use Python SQLAlchemy, Flask, and Rust ORMs and RDBMS technologies to create normalized data structures during ETL. My experience in scale-up and cloud deployment was largely driven by internal protocols and reporting, lift-and-shift deployments, cloud-native or vendor-specific cloud stacks such as AWS, GCP, and Azure. My ETL code is mostly Python, numpy, CUDA, and/or `rust` where I get the time to develop robustly and comfortably at my knowledge of static type function signatures and handling types robustly with GPT4.0-o1-mini in `zed` or Claude Sonnet 3.5+ mobile to assist with code when needed. 

I use minimal software stacks and don't do much UI development, but I'm no stranger to Javascript and CSS frameworks. I don't use Vue.js or Next or Nuxt or MEAN or MERN in most of my app programming. I'm just not a front end guy, I like systems that allow me to look at large sets of variables in a flexible way w.r.t. number of samples and necessary EDA and diagnostic outputs.

I'm a backend programmer and ETL and Data Science are my most accomplished skillset in terms of what I *can* do. But often, you'll want a 'full-stack developer' in the classical sense too. 



## Closing remarks on team skills

  
  With five years in industry bioinformatics and software engineering teams I have expertise in essential knowledge-economy skills of time-management, project management, public speaking, and the technical accumen needed to deliver value for your team.
  

## I have some knowledge:

With deep and well-rounded expertise in molecular biology, genetics, data analytics, and statistics, I have affected several teams and their publications, screening methods, and more. My team oriented style begins with a fundamentals first approach to discussing expertimental or analytical concerns with the Principal Investigator. When artifacts or biophysical aspects of assay design require special treatment, my backgorund in chemistry, biology, mathematics, and data analytics allows me to quickly integrate suggestions from team members into the analytical process, such as filtering bad leads or integrating additional knowledge for specific cases within large datasets.



# Contact

## Do you want to connect?

If you have a research question or you would like to discuss wet-bench, software, or bioinformatics please let me know. 

If you are a recruiter please provide a location, time-frame to hire, and a full job description as plain-text in an email to my [gmail](mailto:mralston.development@gmail.com).

If you are an interviewer and would like to schedule a phone or zoom interview, please copy and paste the following text into your email.

```
name: Jane Doe
email: jane.doe@example.com
company: Example dot com
department: Data engineering
subject: I'd like to contact you about the following opportunity
date: Date (- of opportunity)
time: time of interview
description: A description of the position or opportunity you'd like to disclose to me.
question: Key questions I should address in my response 
kpi: What are your indicators of whether or not this is successful, like a successful interview shedule by a certain date. A fee range you'd like to remain within. What makes our relationship successful and how may I discuss details more
```

----------------------------------------------------------------------




## Testimonials

When asked for references, I've received the following responses from former managers, indicating their support and continued pen-pal relationship during my early career.

Anonymous, Associate Director at Bristol-Myers Squibb

> Whatever you need, big guy.

Anonymous, former boss and mentor at BMS

> Of course. Any time. You don't even need to ask.

## References

```bash
# boss, manager, and program director at Bristol-Myers Squibb
Mark Russo PhD | russomf@hotmail.com
# Committee member and Program Director at Univ. of Delaware
Cathy Wu PhD | wuc@udel.edu
# manager and microbial genomics cloud computing expert at Bayer Crop Sciences
Kyle Lambert | kjlambert@ucdavis.edu
```
