---
permalink: /biosciences
title: "Life science professional"
author: ralston
date: 2025-03-15 03:44:00 +0400
layout: post
categories: [About, Biology]
tags: [About]
render_with_liquid: false
hero_image: "images/dna1.gif"
---

<!-- include page_header.md  -->


Experience with laboratory techniques such as Western/Northern blot, RT-PCR, GC-MS, HPLC, ligation, restriction enzymes, ELISA, IHC/IF, FACS-Aria, Illumina library preparation. You name it!
 
 > Target audience: life scientists, molecular biologists, chemists, bench scienctists, and management professionals
 
## Life science experience
 
If you are looking for a experienced molecular biologist, analytical chemist, and scientist with a balance of laboratory skills and statistics/analytical experience, look no further.
  
My experience with laboratory techniques begins with my Bachelor of Science (BSc Biochemistry, 2012) at the University of Delaware and the Delaware Biotechnology Institute where I performed Northern/Western blots, RT-PCR, Illumina RNA-seq, ELISA, cell culture, IHC, microscopy and many other techniques in GLP/aseptic environments.
 
My research included the prediction and validation of functional RNAs and elucidating their roles in both human/mammalian cells and in bacteria. I worked in areas as a generalist in 2 labs: the Helen Graham Cancer Center's Center for Translational Cancer Research (CTCR, 2009-2012) and the Delaware Biotechnology Institutes laboratory for biofuels research under renowned chemical engineer, genetic engineer, and metabolic engineer Terry Papoutsakis. 
 
 > While there, I performed the first strand-specific RNA-seq experiment in the *Clostridia* genus.
 
 I used the Ribo-Zero kit and Epicenter's ScriptSeq Illumina RNA-seq preparation kit to prepare RNA samples from aliquots sampled from my bioreactors to cross-validate novel RNA species, riboswitches, cis-antisense RNAs, lncRNAs, mRNAs, and other molecular species of the transcriptome, to then additionally validate by Northern blot.

## Cancer research

While at the Helen Graham Cancer Center I studied the anti-cancer effects of statins contributing to the TGF-B and BMP families of SMAD cascades resulting in caspase-mediated apoptosis. I added to the understanding of the lab through the use of immunohistochemistry (IHC) of fresh and FFPE samples, immunofluorescence microscopy, aseptic/sterile colon-cancer cell line culture, Western blot, and qRT-PCR analysis of high-quality RNA samples. 

My extent of cancer familiarity includes many canonical molecular signal transduction pathways, the molecular mechanisms of activation, protein networks, and Illumina RNA-seq and miRNA-seq on hg19 and hg38 transcriptomes.


## Pharmaceutical industry

While in the pharmaceutical industry, I contributed to a variety of disease areas and research topics including RNA-seq, sequencer automation, target discovery, siRNA off-target thermodynamics analysis, ligand preparation and molecular dynamics simulation data pipelines. 

Specifically, I used computational tools and molecular biology knowledgebases to generate leads in a discovery research environment. While there, I contributed to reproducibility research initiatives, and received and award for fast turnaround time regarding a reproducibility frameworks for scientists to use to ensure that calculations and datasets were logged and versioned for accuracy and reprodubility during one of the "checkmate" trials at Bristol-Myers Squibb.
 
# Training

> Univ. of Delaware B.Sc. Biochemistry (2012) and M.Sc. Bioinformatics and Computational Biology (3.96 GPA, 2014)

### Undergraduate program

I developed laboratory fundamentals as a wet-bench laboratory biochemist and molecular biologist. I learned aseptic culture, some immunological, and fluorescent assay techniques, and RT-PCR in a translational cancer laboratory. The UD chemistry department emphasized other laboratory fundamentals such as HPLC, GC-MS, spectrometry, and organic synthesis.

Subects studied include organic chemistry, physical chemistry and thermodynamics, and biochemistry, molecular biology, genetics, physiology, microbiology, virology, physics, linear algebra, probability, calc I/II/III and others.

### Graduate program

My thesis was hybrid wet-lab and dry-lab 2-year project developing the first strand-specific Illumina NGS map of the *C. acetobutylicum* transcriptome during solventogensis and sporulation stresses. Owing of course to the help of Drs. Papoutsakis and Venkataramanan, I investigated the whole-transcriptome response of a biofuel-producing bacterium to the biophysical stresses of solvent/biofuel accumulation and associated stress-response adaptations in anaerobic fermentation conditions. I formulated a fixed media, prepared the bioreactors and innocula, sampled across time points for small molecule characterization by HPLC, RNA-quality investigation, subsequent PCR-mediated Illumina HiSeq 2500 library prepration, and delivered my samples to the Delaware Biotech. Institute core for sequencing. 

Afterwards, I developed both genome browser and interactive volcano-plot visualization web-applications for assessing differential expression of transcripts across the time-stress design.

Subjects included in my graduate program include metabolic engineering, bioinformatics, biostatistics, computational biology, database systems, algorithm topics, sequence alignment, Illumina NGS, biostatistics, Linux HPC training, web application development, Perl, Python and bourne shell (bash) programming. 

## Wet laboratory skills

* Laboratory sciences BSc. (Chemistry department bachelors degree with specialization: **BSc. Biochemistry,** University of Delaware, 2012)

I have demonstrated competencies in many laboratory methods as well as soft skills related to team organization, and I have graduate-level course work and **10 years of industry and academic experience** covering [molecular biology, **human and mammalian genomics and immortalized cell-line culture,** mammalian physiology, microorganism genomics (bacterial, viral) metagenomics and phylogenomics, biofuels and cell culture bioreactor,] process optimization, [novel methodologies and protocols, **Illumina RNA-seq library preparation, qRT-PCR, RT-PCR, PAGE (Western blot, Northern blot), ELISA, Chromatography (TLC, HPLC), Mass Spectrometry (GC-MS, MS-MS, LC-MS),** RNA quality analysis/RIN, Nanodrop, nucleic acid extraction, enzymatic treatment], Flow Cytometry/FACS-Aria, [light microscopy, fluorescence microscopy, immunohistochemistry (IHC), immunofluorescence (IF), gene knockdown, cloning,] bioassay development, and additionally [media formulation, contaminant testing and detection], [technical and software expertise], **[GLP, ELN],** and additional undergraduate lab and bioscience, [natural sciences, undergraduate mathematics, probability, statistics, molecular diversity], [data standards and reporting, **technical writing,** trends and conventions in biodata standards]. 

>NOTE: Skills listed above in brackets indicate genuine industry experiences and/or experience at a professional level.



