---
permalink: /software
title: "10 years of software development experience"
author_profile: true
toc: true
---




With a deep passion for software development and data-driven problem-solving, I specialize in R, Python, and Linux to build efficient, scalable, and robust solutions. My expertise lies in developing data-centric applications, automating workflows, and optimizing system performance for research, analytics, and business intelligence.

Proficient in Python, I design and implement machine learning models, data pipelines, and backend services, ensuring efficiency and maintainability. In R with Bioconductor, I leverage statistical computing and visualization techniques to extract meaningful insights and develop analytical tools tailored to various industries. My fluency in Linux allows me to manage system environments, deploy applications, and automate processes, ensuring seamless integration across platforms.

I thrive in collaborative and innovative environments, constantly refining my skills to tackle complex challenges. Whether it's scripting in Bash, optimizing code performance, or architecting scalable solutions for cloud infastructure using Docker, I am committed to building software that drives impact.

I am eager to contribute my technical expertise and problem-solving mindset to projects that push the boundaries of technology and data science.



# Work experience in bioinformatics software development


My experience spans multiple roles in genomics, bioinformatics, and computational chemistry, with a strong focus on software development, data analysis, and cloud computing. At Bayer Crop Sciences (June-December 2020), I worked as a Genomics Pipeline Developer, leveraging AWS tools and custom in-house software to optimize and scale bioinformatics pipelines. My contribution included expanding a pipeline from five to twenty-four modules within six months, introducing advanced capabilities such as biosynthetic gene discovery, phylogenetic annotations, and genome completeness metrics. Additionally, I developed custom data parsers and json-schema specifications in Python.

During my tenure at Bristol Myers Squibb (2015-2019), I served as a Research Scientist II in Genomics and Computer-Assisted Drug Design (CADD). My work encompassed bioinformatics server administration, sequencer-to-data-pipeline automation, and AWS integration. I developed expertise in RNA sequencing technologies, dataset quality control, and statistical modeling while also creating a version control system for large deduplicated datasets. In the CADD domain, I enhanced a theoretical chemistry framework for molecular enumeration and parameter optimization across multiple chemotypes, refining computational drug discovery methods and time to discovery.

As a Graduate Research Assistant at the Papoutsakis Laboratory (2012-2015), I focused on microbial genomics and transcriptomics, using bioreactors for bacterial fermentation and RNA extraction. I designed a protocol for generating strand-specific Illumina RNA-seq libraries and ensured RNA quality with analytical checkpoints. My computational contribution included building a bioinformatics pipeline to process 1.5 billion paired-end Illumina reads, a custom genome browser written in Python Django and javascript. I produced a transcriptome assembly from diverse culture conditions. Additionally, I supported a research team with computational analyses to advance microbial genomics studies.

Overall, my skill set includes expertise in bioinformatics, software engineering, cloud computing (AWS), and computational drug design. I am proficient in multiple programming languages (Python, R, NodeJS, and SQL) and have experience with various computational frameworks and tools, including Schrodinger, Openeye, and ChemAxon. My contributions across academia and industry reflect a strong ability to integrate software development with biological and chemical research, optimizing data pipelines and computational workflows to advance scientific discovery.



# Skills Developed


* Programming:

Python, R, bash, Javascript/NodeJS, Rust, Emacs, D3.js, Perl, awk, grep/sed, LaTeX, Matlab, HTML/CSS, SQL, SAS


* Relational Database Management:

Oracle SQL, PostgreSQL, MariaDB, SQLite, MongoDB, SQLAlchemy

* Cloud Computing:

AWS, S3, Elastic Compute (EC2), EBS, CodeBuild, Elastic Container Repository (ECR), Docker, IAM

* Statistics/Data Science: 

ANOVA/regression, distribution fitting, R/Rstudio/Rshiny, Bioconductor, discrete/continuous probability, multivariate statistical analysis, PCA, clustering, classification


* Systems Administration: 

systemd, tmux, Nginx, Apache, web servers, Sun Grid Enginer (SGE/UGE), bash, parallel
