




## Who am I?

I'm Matt Ralston and I'm a data science, scientific computing, devops, and knowledge-economy professional. If you want to get to know me, please check out the various topics you can find throughout my website on the [/about](/about) me page.

### Matt in a nutshell

I work in groups of engineers, analysts, and wet-bench laboratory professionals, team leads, associate directors, and alike. I listen to as many voices as I can, and I share my experiences with others; I am detail oriented and I enjoy the problem solving process, and I can discuss your needs in a variety of ways going forwards.

* - has soft skills, style and brevity in communication
* - track-record and wide technical expertise in engineering, chemistry, biology and genomics, bioinformatics, and Illumina WGS/WES/RNAseq/metagenomics
* - knowledge of agile methodology
* - system design mindset: design, benchmarking and testing
* - cloud-native and Docker containers

### Go-to menu | statements considered harmful

* 
* - skip to [\[Communications and documentation\]](#communications-and-documentation)
* - skip to [\[Task and time management\]](#regarding-task-and-time-management)
* - skip to [\[server code vs UI/UX\]](#software-for-backend-and-server-code-vs-UI-and-UX)


## Life science specialty

And, so, I do wet-laboratory science as well. I'm effecient and effective and timely in the lab. Asepsis leads to less errors, and as a bench molecular biologist I'm familiar with the autoclave and where the imaging equipment is. I can perform single-gene targetted expression studies (sometimes referred to as qPCR or rt-PCR on mRNA molecules. I also work with Illumina lab reagent kit technologies). 

## Disclaimer and software licensing

I am certainly **not** an *expert* at modeling, data-processing, and visualization techniques such as *R*, *Cython*/*Python*, *Javascript*, *Linux*/*bash*, *Docker*, *AWS*/cloud and more. But It's more than just websites, APIs, numbers and figures here; my strong fundamentals and data-driven design strategy have been employted towares diverse goals in industry: hypothesis generation, key-questions and KPIs, application or integration needs, and more, such that my products are reproducibile, documented, and clearly written. I can adapt to your team.


I am *familiar* with open source standards, commercial licensing and procurement, software licenses, and EULAs.

And god willing I do multiplex in the lab and on the computer.

I have some conceptual familiarity with a lot of different methods from chemical analytical, bioassay, immunoassay, fluorimetry, MS, GC-MS, LC-MS, ELISA, and an intimate understanding of cell culture and laminar flow hoods. 

My last acura was an Illumina but my last published report was validations of Northerns and novel lnc-RNAs, antisense RNAs, RNAi, and Illumina transcriptomics with strand-specific protocols.


## Communications and Documentation

I work independently with consistent email and Slack communications, and project updates and logging via Version Control and `git`. My project management software is often plain text, also git-based, metrics, org-files for Pandoc compatibility, second-brain, markdown where possible, license documents, source files and configurations, Dockerfiles where possible, and I write LaTeX primarily in `.Rmd`.

Committed to continuous learning and staying updated with the latest advancements in these fields, my breadth of knowledge allows me to model, program, and problem solve in engineering teams. With my natural science knowledge I can ask intelligent questions about variables and covariates, treatment conditions, and their effects in your experimental design. I write clearly in technical and team messaging. 

I have stylistic Javascript and UI/UX ecosystem and preference set and functional and OOP style knowledge. I can write CSS/HTMX, and I can build web apps in any language. I'm very partial to having a static asset stack where possible, Nginx, and I like Github Pages and Jekyll CMS. Most of my content is Markdown, plain-text, and org-mode when it's not software projects.

That being said, most of my technical reporting is LaTeX, Pandoc, and Rmarkdown for scientific documentation.

I use Sphinx and Markdown for most simple code-and-a-story literate programming.

I am no stranger to Flask, FastAPI, Dask, OpenMPI, LAPACK, Tableau, Jupyter, and other standard app-and-reporting codebase trends and ease-of-use and lifetime centric development.

I've worked in Agile teams and I am familiar with scrum, sprints, stories and other development team concepts.

I've used Slack, Skype, and Atlassian distributed-team software and I am consistent on issues tracking, reporting, kanban, application lifetime management suggestions, and of course version control and repository feng-shui.




## GTD/Productivity mindset


Productivity is a modest topic for discussion and I find it enjoyable because you see others perspectives on engineering concerns, assumptions, testing needs, Blue-Green workflows, Red-team and Green-team development, and of course GTD methods when you're with new colleagues and their perspectives.

### Regarding task and time management

My workflow centers on the `README.md`, software usage patterns and documentation, Sphinx, Oxygen, and Pandoc documentation, literate programming, org-mode, and Emacs.

I use the zed editor, PyCharm, Emacs and vi, tmux, and the bash shell for my development environment and I focus heavily on commit messsages, gitflow, rebase/merge/squash source tree design, and I use plain text and org-mode to create and manage my TODO lists and organize information in a Pandoc compatible way.

My task and time management largely revolves in Atlassian and Github kanban and issue tracking systems. I also can use Emacs functionality to perform clock-ins on tasks and git to manage versions of documentation and code, in source trees, to perform time estimates and prototype cost estimation.

I am flexible regarding your task and time tracking and coding requirements speicified by your org chart and PM guidelines.


### Software for backend and server code vs UI and UX

I use Rmarkdown and Python for most of my direct data science stack. I use Python SQLAlchemy, Flask, and Rust ORMs and RDBMS technologies to create normalized data structures during ETL. My experience in scale-up and cloud deployment was largely driven by internal protocols and reporting, lift-and-shift deployments, cloud-native or vendor-specific cloud stacks such as AWS, GCP, and Azure. My ETL code is mostly Python, numpy, CUDA, and/or `rust` where I get the time to develop robustly and comfortably at my knowledge of static type function signatures and handling types robustly with GPT4.0-o1-mini in `zed` or Claude Sonnet 3.5+ mobile to assist with code when needed. 

I use minimal software stacks and don't do much UI development, but I'm no stranger to Javascript and CSS frameworks. I don't use Vue.js or Next or Nuxt or MEAN or MERN in most of my app programming. I'm just not a front end guy, I like systems that allow me to look at large sets of variables in a flexible way w.r.t. number of samples and necessary EDA and diagnostic outputs.

I'm a backend programmer and ETL and Data Science are my most accomplished skillset in terms of what I *can* do. But often, you'll want a 'full-stack developer' in the classical sense too. 



## Closing remarks on team skills

  
  With five years in industry bioinformatics and software engineering teams I have expertise in essential knowledge-economy skills of time-management, project management, public speaking, and the technical accumen needed to deliver value for your team.
  

## I have some knowledge:

With deep and well-rounded expertise in molecular biology, genetics, data analytics, and statistics, I have affected several teams and their publications, screening methods, and more. My team oriented style begins with a fundamentals first approach to discussing expertimental or analytical concerns with the Principal Investigator. When artifacts or biophysical aspects of assay design require special treatment, my backgorund in chemistry, biology, mathematics, and data analytics allows me to quickly integrate suggestions from team members into the analytical process, such as filtering bad leads or integrating additional knowledge for specific cases within large datasets.




